packages/odata-v2/src/uri-conversion/get-resource-path.spec.ts

```diff
@@ -66,4 +66,15 @@ describe('get resource path', () => {
       'There are too many key properties. Ignoring the following keys: StringProperty'
     );
   });
+
+  it('URI encodes keys with special characters', () => {
```

Comment:
perhaps can you add one test for the quote encoding, because this was en error in the past - see here for good examples: https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html